resources/providers/gcplib/inventory/provider.go

```diff
@@ -189,10 +258,109 @@ func mergeAssetContentType(assets []*assetpb.Asset) []*assetpb.Asset {
 			item.IamPolicy = asset.IamPolicy
 		}
 	}
-
 	var results []*assetpb.Asset
 	for _, asset := range resultsMap {
 		results = append(results, asset)
 	}
 	return results
 }
+
+// extends the assets with the project and organization display name
+func extendAssets(ctx context.Context, crm *ResourceManagerWrapper, assets []*assetpb.Asset) []*ExtendedGcpAsset {
+	var extendedAssets []*ExtendedGcpAsset
+	var cache = make(map[string]*fetching.EcsGcpCloud)
```

Comment:
We create a new cache for every type, consider storing the cache at the provider level.